class-wc-facebookcommerce.php

```diff
@@ -924,6 +925,22 @@ public function get_profiling_logger() {
 			return $instance;
 		}
 
+		/**
+		 * Get the product sync validator class.
+		 *
+		 * @param WC_Product $product A product object to be validated.
+		 *
+		 * @return ProductSyncValidator
+		 */
+		public function get_product_sync_validator( WC_Product $product ) {
```

Comment:
We can't operate this way.  `get_product_sync_validator` will not create a fresh instance when we will try to validate another product. We are stuck with the original `$product` that was passed to the constructor.
Functions like `add_product_list_table_columns_content` need a new instance each time - used for products table display. Or if we want to keep a single class then we need to pass `$product` to `validate()` and not use it in the constructor.